pkg/processor/runtime/python/py/_nuclio_wrapper_as_server.py

```diff
@@ -0,0 +1,290 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import inspect
+import sys
+import traceback
+import time
+import json
+import nuclio_sdk
+import nuclio_sdk.helpers
+import nuclio_sdk.json_encoder
+import nuclio_sdk.logger
+from wrapper_common import (
+    WrapperFatalException,
+    AbstractWrapper,
+    create_logger,
+    get_parser_with_common_args)
+import socket
+import asyncio
+import selectors
+
+
+class Wrapper(AbstractWrapper):
+    def __init__(self,
+                 logger,
+                 loop,
+                 handler,
+                 serving_address,
+                 control_socket_path,
+                 platform_kind,
+                 namespace=None,
+                 worker_id=None,
+                 trigger_kind=None,
+                 trigger_name=None,
+                 decode_event_strings=True,
+                 max_connections=None):
+        super().__init__(logger, loop, handler, control_socket_path, platform_kind, namespace, worker_id, trigger_kind,
+                         trigger_name, decode_event_strings)
+
+        # Validate that the handler is an async function
+        if not inspect.iscoroutinefunction(self._entrypoint):
+            raise WrapperFatalException(f"The provided handler '{self._entrypoint.__name__}' "
+                                        f"must be an async function (async def).")
+        split_address = serving_address.split(":")
+        self._host = split_address[0]
+        self._port = int(split_address[1])
+
+        # Max file descriptors allowed by the OS
+        self._max_connections = max_connections if max_connections else os.sysconf("SC_OPEN_MAX")
+        self.selector = selectors.DefaultSelector()
+        self.connections = {}  # Track active sockets
+        self.tasks = {}  # Track active asyncio tasks
+        self.shutdown_event = asyncio.Event()
+
+    async def _handle_event(self, event, sock):
+        # take call time
+        start_time = time.time()
+
+        # call the entrypoint
+        entrypoint_output = await self._entrypoint(self._context, event)
+
+        # measure duration, set to minimum float in case execution was too fast
+        duration = time.time() - start_time or sys.float_info.min
+
+        await self._write_packet_to_processor(sock, 'm' + json.dumps({'duration': duration}))
+
+        # try to json encode the response
+        encoded_response = self._encode_entrypoint_output(entrypoint_output)
+
+        # write response to the socket
+        await self._write_packet_to_processor(sock, 'r' + encoded_response)
+
+    async def _process_connection(self, sock):
+        """Process all events for a single connection."""
```

Comment:
Just to make it clear (and maybe add it as a comment) - Once a connection is opened once from the processor side, this task is created and is long running and can accept events one after the other. 
right?

Reply:
yes